packages/solana/lib/chain/SolanaChainProvider.ts

```diff
@@ -0,0 +1,107 @@
+import { Chain } from '@liquality/client';
+import { UnsupportedMethodError } from '@liquality/errors';
+import { AddressType, Asset, BigNumber, Block, FeeDetails, Network, Transaction } from '@liquality/types';
+import { AccountLayout, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import { Connection, PublicKey } from '@solana/web3.js';
+import { parseBlockResponse, parseTransactionResponse } from '../utils';
+
+export class SolanaChainProvider extends Chain<Connection, Network> {
+    constructor(network: Network) {
+        super(network);
+
+        if (!this.provider && this.network.rpcUrl) {
+            this.provider = new Connection(network.rpcUrl, 'confirmed');
+        }
+    }
+
+    getBlockByHash(_blockHash: string): Promise<Block<any, any>> {
+        throw new UnsupportedMethodError('Method not supported for Solana');
+    }
+
+    public async getBlockByNumber(blockNumber?: number, includeTx?: boolean): Promise<Block<Block, Transaction>> {
```

Comment:
Why `_raw` block object also has the same type?

Reply:
Fixed